corehq/apps/users/views/__init__.py

```diff
@@ -207,14 +207,16 @@ def form_user_update(self):
             raise NotImplementedError("You must specify a form to update the user!")
 
         if self.request.method == "POST" and self.request.POST['form_type'] == "update-user":
-            return self.user_update_form_class(data=self.request.POST)
+            form = self.user_update_form_class(data=self.request.POST)
+        else:
+            form = self.user_update_form_class()
+            form.initialize_form(domain=self.request.domain, existing_user=self.editable_user)
 
-        form = self.user_update_form_class()
-        form.initialize_form(domain=self.request.domain, existing_user=self.editable_user)
         if self.can_change_user_roles:
             form.load_roles(current_role=self.existing_role, role_choices=self.user_role_choices)
         else:
             del form.fields['role']
```

Comment:
This used to be applied in a subclass, so it was applied regardless of where the function returned
https://github.com/dimagi/commcare-hq/pull/14534/files#diff-b88c82ab45221007abc47af5a1c62207L281
https://github.com/dimagi/commcare-hq/pull/14534/files#diff-df796c59286ba51267f9d91ce003127aL269